enhancements/security/runtime-class-scc.md

```diff
@@ -0,0 +1,248 @@
+```
+type SecurityContextConstraints struct {
+    ...
+	// EnsuredRuntimeClasses is a list of names of runtime classes (also called runtime handlers).     // The runtime class specified in a pod request must match one of the items in this list.
+	// To ensure any runtime class, '*' may be specified.
+	// +nullable
+	EnsuredRuntimeClasses []string `json:"ensuredRuntimeClasses" protobuf:"bytes,26,rep,name=ensuredRuntimeClasses"`
+    ...
+}
+```
+Fields in this slice will also be logically OR'd. A user must specify an item that only matches one field to satisfy the requirement. This allows an admin to specify a set of runtime classes, of which the user must choose at least one.
+
+### Some Examples
+Here are some RuntimeClasses that will be useful for Openshift to support, as well as reasonining behind not allowing everyone to use them:
+- Builder: will need access to /dev/fuse on the host, and a user namespace allocated. While these features are available to all users on the host, an untrusted user could take up all available UIDs/GIDs, preventing builds from happening.
+- High Performance: Allows for pods to request the system is configured for latency sensitive workloads. It allows for CPU load balancing, CPU quota disabling, and IRQ smp load balancing to be disabled. A malicious user could get improper access to the host hardware if allowed to use this runtime class.
+- Kata: Allows for pods to be run using the kata runtime, as opposed to runc. This runtime class is the most innocuous, as it only makes the container process more secure. However, kata is known to perform slower and use more resources than runc, so it is definitely something the admin should opt-in to. Further, it may be the case that an admin wants some user to only use kata (a really untrusted workload on a very paranoid node). This is the prime example for `EnsuredRuntimeClasses`.
+
+### User Stories [optional]
+
+#### Story 1
+As a cluster admin, I want to require specific users use the kata runtime for their workloads, and I want to block other users from using it.
+
+#### Story 2
+As a cluster admin, I want to enable unprivileged builds by default on the cluster to make my builds more secure.
+
+#### Story 3
+As a cluster admin, I want to enable performance add-on operator to tune my workloads. 
+
+#### Story 4
+As a cluster admin, I want to ensure that my control plane pods only run in native containers.
+
+#### Story 5
+As a cluster admin, I want certain pods to be able to specify a lower memory limit by using crun
+as the runtime instead of runc as crun has less startup memory overhead.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+We don't want to modify the default runtime class (runc) for the existing SCCs
+as part of this first step.
+Unprivileged builds is a case where we may create the SCC by default (whereas the others are added by an operator for specific installations), but that will also be addressed in a follow up and is out of scope.
```

Comment:
can you add the SCC you imagine creating and the related clusterroles and clusterrolebindings?

I suspect that with something like a kata runtime, we will have a new set of SCCs and new set of default bindings and I'd like to understand how you see the clusterroles being structured to fulfill the usual intent of "untrusted users can do this" and "trusted users can do all that".

Reply:
I'd like this PR to only cover the addition of this field to the SCC structure. If we can agree that there's a compelling reason to add this (any of the use cases can be satisfied with this addition), then the nitty gritty of how the existing SCCs will be adapted/what new ones will be added feels out of scope.